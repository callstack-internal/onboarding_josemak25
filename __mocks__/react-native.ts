jest.mock("react-native", () => {
  const RN = jest.requireActual("react-native"); // use original implementation, which comes with mocks out of the box

  // mock modules/components created by assigning to NativeModules
  RN.NativeModules.SourceCode = {
    scriptURL: "http://localhost:3000/",
  };

  return RN;
});
