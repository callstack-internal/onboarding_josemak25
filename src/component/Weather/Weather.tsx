import React from "react";
import {
  Text,
  View,
  Image,
  TouchableOpacity,
  TouchableOpacityProps,
} from "react-native";
import FontAwesome from "react-native-vector-icons/FontAwesome";

FontAwesome.loadFont(); // initialize font icons

import { makeUseStyles } from "../../helpers/makeUseStyles";

type WeatherProps = { data: Weather } & TouchableOpacityProps;

export const Weather: React.FC<WeatherProps> = ({
  data,
  style,
  onPress,
  disabled,
  ...restProps
}) => {
  const { styles, palette } = useStyles();
  const { main, weather, name } = data;
  const [{ description, icon }] = weather;

  return (
    <TouchableOpacity
      onPress={onPress}
      disabled={disabled}
      accessibilityRole="button"
      testID="weather_card_press"
      accessibilityLabel="weather card"
      style={[styles.cardContainer, style]}
      {...restProps}>
      <View style={styles.cardWrapper}>
        <Image
          style={styles.image}
          resizeMode="contain"
          accessibilityIgnoresInvertColors
          source={{ uri: `https://openweathermap.org/img/wn/${icon}.png` }}
        />
        <View style={styles.cardCityNameWrapper}>
          <Text style={[styles.degreeText, styles.weatherText]}>{name}</Text>
          <Text style={[styles.degreeText, styles.weatherText]}>
            {description}
          </Text>
        </View>
      </View>
      <View style={[styles.cardWrapper, styles.moreContainer]}>
        <View style={styles.degreeWrapper}>
          <Text style={styles.degreeText}>{main.temp} Â°F</Text>
        </View>

        {!disabled && (
          <FontAwesome
            size={14}
            name="chevron-right"
            color={palette.text}
            style={styles.moreIcon}
          />
        )}
      </View>
    </TouchableOpacity>
  );
};

const useStyles = makeUseStyles(({ layout, palette, colors, fonts }) => ({
  cardContainer: {
    height: 70,
    flexDirection: "row",
    alignItems: "center",
    paddingHorizontal: layout.gutter,
  },
  cardWrapper: {
    flex: 1,
    alignItems: "center",
    flexDirection: "row",
  },
  cardCityNameWrapper: {
    marginHorizontal: layout.gutter,
  },
  moreContainer: {
    justifyContent: "flex-end",
  },
  image: {
    width: 50,
    height: 50,
  },
  degreeWrapper: {
    borderRadius: 50,
    backgroundColor: palette.degree,
    paddingVertical: layout.gutter / 1.5,
    paddingHorizontal: layout.gutter * 1.5,
  },
  moreIcon: {
    opacity: 0.5,
    marginLeft: layout.gutter,
  },
  degreeText: {
    fontSize: fonts.size.md,
    color: colors.light.white,
    fontWeight: fonts.weight.semi,
  },
  weatherText: {
    opacity: 0.5,
    color: palette.text,
  },
}));
