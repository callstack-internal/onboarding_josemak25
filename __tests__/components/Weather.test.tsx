import React from "react";

import { render, fireEvent, screen } from "@testing-library/react-native";
import { Weather } from "../../src/component/Weather";

export const data = {
  coord: {
    lon: -3.7026,
    lat: 40.4165,
  },
  sys: {
    country: "ES",
    timezone: 7200,
    sunrise: 1666247440,
    sunset: 1666286896,
  },
  weather: [
    {
      id: 803,
      main: "Clouds",
      description: "broken clouds",
      icon: "04d",
    },
  ],
  main: {
    temp: 17.9,
    feels_like: 17.44,
    temp_min: 14.51,
    temp_max: 20.81,
    pressure: 1016,
    humidity: 65,
  },
  visibility: 8000,
  wind: {
    speed: 13.89,
    deg: 210,
  },
  clouds: {
    all: 75,
  },
  dt: 1666271369,
  id: 3117735,
  name: "Madrid",
};

describe("components / Weather.tsx", () => {
  it("should render correctly", () => {
    const onNavigationPress = jest.fn();

    render(<Weather data={data} onPress={onNavigationPress} />);

    expect(screen.getByText(data.name)).toBeTruthy();
    fireEvent.press(screen.getByTestId("weather_card_press"));
    expect(onNavigationPress).toHaveBeenCalled();
    expect(onNavigationPress).toHaveBeenCalledTimes(1);
  });
});
