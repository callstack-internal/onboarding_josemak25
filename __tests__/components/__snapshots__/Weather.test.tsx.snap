// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components / Weather.tsx should render correctly 1`] = `
<View
  accessibilityLabel="weather card"
  accessibilityRole="button"
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "height": 70,
      "opacity": 1,
      "paddingHorizontal": 12,
    }
  }
  testID="weather_card_press"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <Image
      accessibilityIgnoresInvertColors={true}
      resizeMode="contain"
      source={
        Object {
          "uri": "https://openweathermap.org/img/wn/04d.png",
        }
      }
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    />
    <View
      style={
        Object {
          "marginHorizontal": 12,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 16,
              "fontWeight": "600",
            },
            Object {
              "color": "#000",
              "opacity": 0.5,
            },
          ]
        }
      >
        Madrid
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 16,
              "fontWeight": "600",
            },
            Object {
              "color": "#000",
              "opacity": 0.5,
            },
          ]
        }
      >
        broken clouds
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
        },
        Object {
          "justifyContent": "flex-end",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#b4d7e3",
          "borderRadius": 50,
          "paddingHorizontal": 18,
          "paddingVertical": 8,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 16,
            "fontWeight": "600",
          }
        }
      >
        17.9
         °F
      </Text>
    </View>
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 14,
          },
          Object {
            "marginLeft": 12,
            "opacity": 0.5,
          },
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
